### PR TITLE
Store onboarding: Dismiss payment setup web view when setup completes

### LINKED ISSUE #10573: Handle payment setup completion
Detect the URL for the following param to dismiss the web view automatically:
- wcpay-connection-success
- wcpay-connection-error

### PR DESCRIPTION
**Pull request by itsmeichigo**
Closes: #10573 

## Description

This PR updates the handling for payment setup web view to detect the completion and dismiss the web view when appropriate.
- Add a new view model for the setup web view with a callback closure for handling completion.
- Check for the params `wcpay-connection-error` and `wcpay-connection-success` to trigger the completion closure.

## Testing instructions

- Create a new JN site with Jetpack and Woo. Set up your site with a WCPay-supported country (e.g. US).
- Follow the setup steps in P91TBi-4BH-p2 up to the step of installing the dev plugin only.
- Log in to the app and switch to the JN site.
- On the My Store screen, select Get Paid on the onboarding task list.
- Select Continue Setup > Finish Setup.
- Tap the link to go back to WooPayment, notice that the web view is dismissed.
- Tap the Get Paid task again, this time finish the setup in test mode. When the setup completes (the page says "Redirecting to WooPayments..."), the web view should be dismissed.

## Screenshots

When canceling setup:

https://github.com/woocommerce/woocommerce-ios/assets/5533851/f34c1d7d-679e-4e23-91b6-1cbb6aab5e12

When finishing setup:

https://github.com/woocommerce/woocommerce-ios/assets/5533851/8a4cce58-c4f8-4223-ab50-0586f7f09fda

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by itsmeichigo** [comment]
Converted PR back to draft to update the logic following p1693385159319349/1693384002.341619-slack-C03L1NF1EA3.

**Comment by itsmeichigo** [comment]
Keeping the completion trigger following p1693387310539519/1693384002.341619-slack-C03L1NF1EA3.

File: WooCommerce/Classes/ViewRelated/Dashboard/Onboarding/WooPaymentSetupWebViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    func handleRedirect(for url: URL?) {
+        guard let urlString = url?.absoluteString else {
+            return
+        }
+        // If the self-hosted site allows login with WPCOM as a Jetpack feature,
+        // WPCOM authentication is complete after redirecting to the WPCOM homepage.
+        if urlString.removingSuffix("/") == Constants.urlAfterWPComAuth,
+           initialURL?.absoluteString != Constants.urlAfterWPComAuth {
+            reloadWebview()
+        }
```
Comment (selanthiraiyan) [inline WooCommerce/Classes/ViewRelated/Dashboard/Onboarding/WooPaymentSetupWebViewModel.swift]: Nit: We could consider moving this logic inside `AuthenticatedWebViewModel` as a helper function. 

Something like below,
```swift
extension AuthenticatedWebViewModel {
    func shouldReload(for url: URL?) -> Bool {
        // If the self-hosted site allows login with WPCOM as a Jetpack feature,
        // WPCOM authentication is complete after redirecting to the WPCOM homepage.
        let urlAfterWPComAuth = "https://wordpress.com"
        guard let urlString = url?.absoluteString else {
            return false
        }

        if urlString.removingSuffix("/") == urlAfterWPComAuth,
           initialURL?.absoluteString != urlAfterWPComAuth {
            return true
        }
        return false
    }
}
```

**Review by selanthiraiyan — APPROVED** [review]
This works as expected. 🚀

File: WooCommerce/Classes/ViewRelated/Dashboard/Onboarding/WooPaymentSetupWebViewModel.swift (hunk 1 above)
Comment (itsmeichigo) [reply WooCommerce/Classes/ViewRelated/Dashboard/Onboarding/WooPaymentSetupWebViewModel.swift]: Good idea! Added the extension method in ef867d45d3418030974703cde7631085e1be1074 and updated the web view models accordingly.

**Pull request merged**